### PR TITLE
Fix some golint failures of pkg/kubelet

### PR DESCRIPTION
**Pull request by SataQiu**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Fix golint failures of pkg/kubelet

**Which issue(s) this PR fixes**:

Ref #68026

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm

Thank you for tackling this!

**Comment by yujuhong** [comment]
/lgtm
/approve

Thanks!

File: pkg/kubelet/reason_cache.go [hunk 1]
```diff
@@ -40,8 +40,8 @@ type ReasonCache struct {
 	cache *lru.Cache
 }
 
-// Reason is the cached item in ReasonCache
-type reasonItem struct {
+// ReasonItem is the cached item in ReasonCache
+type ReasonItem struct {
```
Comment (liggitt) [inline pkg/kubelet/reason_cache.go]: Do we need to export this? Can we unexport ReasonCache instead?

Comment (yujuhong) [reply pkg/kubelet/reason_cache.go]: Good catch.

I forgot that ReasonCache is not in a separate subpackage. 
+1 for unexporting it.

Comment (SataQiu) [reply pkg/kubelet/reason_cache.go]: Thanks @liggitt @yujuhong 
But `NewReasonCache` wants an exported `ReasonCache`.
```
// NewReasonCache creates an instance of 'ReasonCache'.
func NewReasonCache() *ReasonCache {
	return &ReasonCache{cache: lru.New(maxReasonCacheEntries)}
}
```
They are very difficult to satisfy at the same time, so I won't modify it for now.

Comment (SataQiu) [reply pkg/kubelet/reason_cache.go]: Can you review this one again? @yujuhong

**Comment by SataQiu** [comment]
Can you have a look at this? @yujuhong

**Comment by yujuhong** [comment]
/lgtm

I'm okay with merging the existing change first.

**Pull request merged**